### ISSUE TITLE
Multipart file not deleted when client aborts upload

### ISSUE THREAD

**Issue by jo-pol** [eclipse/jetty.project#3683]
We've studied #2562 and took a dive into the code but were not able to locate our problem.

When we let a client cancel a multipart upload, jetty does not delete the partial received/saved multi-part file. Any suggestions? 

I attached debug logging, which confirms we used build 9.4.18.v20190429
[early-eof.txt](https://github.com/eclipse/jetty.project/files/3202565/early-eof.txt)

cc @rvanheest

**Comment by joakime** [comment]
When you say "let a client cancel a multipart upload", do you mean the client didn't finish sending the multipart and/or the connection was terminated?

**Comment by janbartel** [comment]
I'm not seeing any debug output that I'd expect from the org.eclipse.jetty.http.MultiPartParser, which is odd. I think you're also not using the default logging setup, because I can't see the names of the threads in the log output, so it's a bit more difficult to see what is going on. 

 Can you set the server to do a dumpAfterStart, and also attach that, so we can see the configuration?

**Comment by jo-pol** [comment]
With cancel I mean use a cancel button if provided, otherwise just close the page/browser.

While diving into this issue I discovered `NCSARequestLog` and got it working in another [branch](https://github.com/DANS-KNAW/easy-deposit-api/pull/153) than used to create attached logging. Tried but did not yet succeed in following the subsequent leads to replace it with the logback configuration of our own logging. 

I'm out of reach of a development environment for the next 36 hours. Then I'll try the `dumpAfterStart`.

**Comment by jo-pol** [comment]
A new [log.txt](https://github.com/eclipse/jetty.project/files/3210912/startup-log.txt) created with https://github.com/DANS-KNAW/easy-deposit-api/blob/2d911455469666e51bbe761c9afb99bc9af19779/src/main/scala/nl.knaw.dans.easy.deposit/EasyDepositApiService.scala#L53-L60

does that help or am I still missing some configurations?

**Comment by lachlan-roberts** [comment]
I don't think the `RequestLog` will help much, I think the DEBUG logging should be more useful.

But there is no MultiPart logging because by default the LEGACY MultiPart compliance mode is used, so it will be using the `MultiPartInputStreamParser` which has no DEBUG logging.

You can use the new RFC7578 compliance mode which will use the new `MultiPartFormInputStream` and `MultiPartParser`, this would give some debug logging during the parsing of the request.

You can do this by setting the compliance mode on the `HttpConfiguration`
```
connector.getConnectionFactory(HttpConnectionFactory.class).getHttpConfiguration()
.setMultiPartFormDataCompliance(MultiPartFormDataCompliance.RFC7578);
```

However the new MultiPart parser may not have the same issue but it would be interesting to find out.
I will also attempt to replicate the issue locally.

**Comment by jo-pol** [comment]
`RequestLog` is just a side trac. Discovered the feature while debugging this issue.

I'm not sure I [implemented](https://github.com/DANS-KNAW/easy-deposit-api/blob/beaeb9bb5d974e2c3e8e700161bdbbe77b606c64/src/main/scala/nl.knaw.dans.easy.deposit/EasyDepositApiService.scala#L34-L39) your advice correctly. I still don't see `MultiPartFormInputStream` but I do see `MultiPartParser` in the [logging](https://github.com/eclipse/jetty.project/files/3211644/early-eof-log.txt) The multipart file of cancelled uploads are still not removed.

**Comment by jo-pol** [comment]
Sorry, now there actually is one log line: 
`DEBUG headerComplete org.eclipse.jetty.http.MultiPartFormInputStream$Handler@4076675e`

**Comment by lachlan-roberts** [comment]
Yes this has been done correctly `MultiPartFormInputStream` does not have many DEBUG logs either but uses the `MultiPartParser`. In your log files the class and thread names seem to be missing which make it quite hard to follow.

example from default jetty logging:
```
2019-05-23 20:41:15.351:DBUG:oejut.ReservedThreadExecutor:main: ReservedThreadExecutor@30c15d8b{s=0/4,p=0}
2019-05-23 20:41:15.351:DBUG:oejuc.ContainerLifeCycle:main: QueuedThreadPool[qtp350068407]@14dd9eb7{STARTING,8<=0<=200,s=0,i=0,r=-1,q=0}[ReservedThreadExecutor@30c15d8b{s=0/4,p=0}] added {ReservedThreadExecutor@30c15d8b{s=0/4,p=0},AUTO}
2019-05-23 20:41:15.352:DBUG:oejuc.AbstractLifeCycle:main: starting ReservedThreadExecutor@30c15d8b{s=0/4,p=0}

```

**Comment by jo-pol** [comment]
Somehow my attempt to add class names was lost. Now I configured with logback:
```
<pattern>[%date{ISO8601}] %-5level [%thread] %logger{10} [%file:%line] %msg%n</pattern>
```

Not exactly the same format but getting close: [early-eof-log.txt](https://github.com/eclipse/jetty.project/files/3212097/early-eof-log.txt)

**Comment by lachlan-roberts** [comment]
@jo-pol can you confirm whether these changes have fixed the problem for you

**Comment by jo-pol** [comment]
Yes. The problem is fixed. Thanks.
